src/common/root/index.js

```diff
@@ -1,111 +1,70 @@
 /* global fetch:false */
-import React, { Component } from "react";
+import React, { useState, useEffect } from "react";
 import debounce from "lodash.debounce";
 import Linter from "../linter";
 import recommendedConfig from "stylelint-config-recommended";
 import standardConfig from "stylelint-config-standard";
 import "whatwg-fetch";
 
 const defaultCSS = "a {color: #FFF; }\n";
-const config = {
+const defaultConfig = {
   rules: Object.assign(recommendedConfig.rules, standardConfig.rules)
 };
 
-export default class Root extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      code: defaultCSS,
-      config: JSON.stringify(config, null, 2),
-      syntax: "css",
-      warnings: [],
-      error: false
-    };
-    this.lint = debounce(this.lint, 250);
-    this.lint = this.lint.bind(this);
-    this.onCodeChange = this.onCodeChange.bind(this);
-    this.onConfigChange = this.onConfigChange.bind(this);
-    this.onSyntaxChange = this.onSyntaxChange.bind(this);
-  }
-
-  componentDidMount() {
-    this.lint();
-  }
+export default function Root() {
+  const [code, setCode] = useState(defaultCSS);
+  const [config, setConfig] = useState(JSON.stringify(defaultConfig, null, 2));
+  const [syntax, setSyntax] = useState("css");
+  const [warnings, setWarnings] = useState([]);
+  const [error, setError] = useState(false);
 
-  lint() {
+  function lint() {
     fetch("/lint", {
       method: "POST",
       headers: {
         Accept: "application/json",
         "Content-Type": "application/json"
       },
       body: JSON.stringify({
-        code: this.state.code,
-        config: this.state.config,
-        syntax: this.state.syntax
+        code,
+        config,
+        syntax
       })
     })
       .then(response => {
         return response.json();
       })
       .then(data => {
         if (data.error) {
-          this.setState({
-            error: data.error
-          });
+          setError(data.error);
         } else {
-          this.setState({
-            warnings: data.warnings,
-            error: false
-          });
+          setWarnings(data.warnings);
+          setError(false);
         }
       })
       .catch(error => {
-        this.setState({
-          error: `Unable to lint CSS: \n\n ${error}`
-        });
+        setError(`Unable to lint CSS: \n\n ${error}`);
       });
   }
 
-  onCodeChange(code) {
-    this.setState(
-      {
-        code
-      },
-      this.lint
-    );
-  }
-
-  onConfigChange(config) {
-    this.setState(
-      {
-        config
-      },
-      this.lint
-    );
-  }
+  useEffect(() => {
+    debounce(lint, 250)();
+  });
```

Comment:
[question] If calling `debounce` outside `useEffect`, do you think there is an advantage on performance?

For example:
```js
const debounceLint = debounce(lint, 250);
useEffect(() => {
  debounceLint();
});
```

Reply:
I did just like that initially but then thought why to declare a variable for a single use :) I don't know if there any performance differences. I think it's even possible to simplify to the following:

```js
useEffect(debounce(lint, 250));
```

But it might be confusing to understand what is actually this code does.

Reply:
> I don't know if there any performance differences.

It might be no difference as I mind. Anyway, nobody knows it unless measuring the performance actually. 🤷‍♂️ 

> But it might be confusing to understand what is actually this code does.

Yeah, this seems to be confusing to me too. 😅

Reply:
Thanks a lot for your response!